Tweaked native tokens explanation with more details

olgahryniuk opened the pull request:
@JaredCorduan 
This PR adds the content which we have on Cardano Docs (such as a getting started tutorial, native tokens exercises, exchange multi-asset management scenarios, and listing advice) and also includes some styling tweaks to existing sections. Can you please review and merge if happy? I also made changes to the rst index file so that new sections are included in the user-facing documentation.

olgahryniuk committed:
Create learn-about-native-tokens.rst

olgahryniuk committed:
Merge pull request #1 from input-output-hk/olgahryniuk-patch-1

Create learn-about-native-tokens.rst

olgahryniuk committed:
initial review

olgahryniuk committed:
Update token-bundles.rst

olgahryniuk committed:
final styling tweaks

olgahryniuk committed:
Adding more about ERC20 comparison

olgahryniuk committed:
adding details about the native token lifecycle

olgahryniuk committed:
Update min-utxo-mary.rst

olgahryniuk committed:
Update min-utxo-alonzo.rst

olgahryniuk committed:
Create get-started.rst

This is a step-by-step manual for working with native tokens. Last reviewed by IOG QA team before the Vasil release. All commands are updated to match new environments.

olgahryniuk committed:
Create exercises.rst

olgahryniuk committed:
Update faq.rst

olgahryniuk committed:
Create exchange-scenarios.rst

olgahryniuk committed:
Create listing.rst

olgahryniuk committed:
Update index.rst

JaredCorduan commented on doc/explanations/exchange-scenarios.rst:
```diff
@@ -0,0 +1,257 @@
+
+   curl http://localhost:8090/v2/byron-wallets/ 
+
+**Shelley**
```
```suggestion
**Shelley and beyond**
```

JaredCorduan commented on doc/explanations/token-bundles.rst:
```diff
@@ -94,19 +103,18 @@ This is an example of a token bundle, let’s call it ``TB_Example`` : ::
 ********************************************
 
-Token bundles can be found:
+Token bundles can be found: 
```
```suggestion
Token bundles can be found:
```

olgahryniuk committed:
Update exchange-scenarios.rst

JaredCorduan commented on doc/explanations/exchange-scenarios.rst:
```diff
@@ -0,0 +1,244 @@
+=========================================
+Exchange multi-asset management scenarios
```
This looks like some really excellent documentation, but it is uses a lot of components downstream of the ledger:
* cardano-graphql
* cardano-rosetta
* cardano-db-sync
* cardano-wallet

The folks working in this repository (cardano-ledger) will not know when things fall out of date, and this documentation will very likely quickly become invalid if it lives here.

olgahryniuk replied:
Noted, thanks! I'll delete this explainer from the ledger repo and see where we can host it.

JaredCorduan commented on doc/explanations/exercises.rst:
```diff
@@ -0,0 +1,104 @@
+=======================
+Native tokens exercises
```
This documentation should probably live in the cardano-node repository.

olgahryniuk replied:
Noted, thanks! I'll delete it from here and talk to the node team to host this in 'references'

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
+field) of the transaction. If the quantity of a particular asset in the
+mint field is negative, the result is that after processing the
+transaction, the total quantity of that specific asset on the ledger
+will be reduced by the amount reflected in the mint field.
```
```suggestion
A. Every ledger era starting with Mary contains a field in the transaction
body for minting multi-assets, named the mint field.
In order to use the minting field, transaction must be authorized by the
minting policy. Positive values create assets, and negative ones destroy them.
```

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
+signatures. Transaction data must satisfy all the minting policies of
+all assets the transaction is attempting to mint.
+
+**Q. What is a token builder and what is its functionality?**
```
The token builder is not a part of the ledger, so this explanation will need to live elsewhere, unfortunately.

olgahryniuk replied:
Deleted, thanks!

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
 
-The multisig scripting language (which existed prior to the introduction of MA functionality on Cardano) specifies some minimal set of signatures required to allow a transaction to perform a certain action, usually to spend a UTxO entry.
+A. The multisig scripting language (which existed prior to the
+introduction of MA functionality on Cardano) specifies some minimal set
```
the multi-sig DSL was actually introduced in the Shelley ledger era.

olgahryniuk replied:
Tweaked to avoid inaccuracy

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
+A. In a one-time minting policy, the complete set of tokens scoped under
+it is minted by one specific transaction. This means that no more tokens
+will ever be minted under that policy. This type of policy does require
+smart contracts and cannot be expressed using multisig.
```
The multi-sig DSL was expanded in the Allega ledger era to include timelocks, which do provide the ability to guarantee that a given policy can never again be used after a gives slot.

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
-
-A mixed collection of tokens scoped under one or more minting policies. Any tokens can be bundled together.
+**Q. What is a token bundle?** A. A mixed collection of tokens scoped
+under one or more minting policies. Any tokens can be bundled together.
```
do you want to mention asset IDs?

olgahryniuk replied:
I think we have a nice explainer in the dedicated section (assets, tokens, and token bundles). No need to write too much here.

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
 
-How do native tokens appear in a user's wallet?
-********************************************************************************
+**Q. How do native tokens appear in a user’s wallet?**
```
there is no notion of a "wallet" in the ledger, so this description should not live here.

olgahryniuk replied:
deleted

JaredCorduan commented on doc/explanations/faq.rst:
```diff
@@ -1,218 +1,337 @@
 
-How does coin selection work for custom native tokens?
-********************************************************************************
+**Q. How does coin selection work for custom native tokens?**
```
coin selection really has nothing to do with the ledger, but really a wallet implementation detail.

olgahryniuk replied:
deleted

JaredCorduan commented on doc/explanations/get-started.rst:
```diff
@@ -0,0 +1,719 @@
+==================================
+Getting started with native tokens
```
This documentation should probably live in the cardano-node repository.

olgahryniuk replied:
deleted

olgahryniuk committed:
Update doc/explanations/exchange-scenarios.rst

Co-authored-by: Jared Corduan <jaredcorduan@gmail.com>

olgahryniuk committed:
Update doc/explanations/token-bundles.rst

Co-authored-by: Jared Corduan <jaredcorduan@gmail.com>

olgahryniuk committed:
Update doc/explanations/faq.rst

Co-authored-by: Jared Corduan <jaredcorduan@gmail.com>

olgahryniuk committed:
delete exercises

olgahryniuk committed:
Delete exchange-scenarios.rst

olgahryniuk committed:
delete 'what is a token builder?'

olgahryniuk committed:
tweak multisig wording

olgahryniuk committed:
delete the wallet question

olgahryniuk committed:
delete 'coin selection'

olgahryniuk committed:
Delete get-started.rst

olgahryniuk committed:
Update index.rst

olgahryniuk commented:
@JaredCorduan I think I have now addressed all your suggestions. Thanks so much for going through this! Please take a look if everything is okay now and if the index.rst is properly edited.

JaredCorduan reviewed (approved):
looks great, thank you @olgahryniuk !
